src/evaluate/hub.py

```diff
@@ -0,0 +1,125 @@
+from typing import Dict
+
+import requests
+from datasets.utils.metadata import known_task_ids
+from huggingface_hub import dataset_info, model_info
+from huggingface_hub.repocard import metadata_update
+
+from .utils.logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+def get_allowed_tasks(tasks_dict):
+    return list(tasks_dict.keys())
+
+
+def push_to_hub(
+    repo_id: str,
+    task_type: str,
+    dataset_type: str,
+    dataset_name: str,
+    metric_type: str,
+    metric_name: str,
+    metric_value: float,
+    task_name: str = None,
+    dataset_config: str = None,
+    dataset_split: str = None,
+    dataset_revision: str = None,
+    dataset_args: Dict[str, int] = None,
+    metric_config: str = None,
+    metric_args: Dict[str, int] = None,
+    overwrite: bool = False,
+):
+    r"""
+    Pushes the result of a metric to the Hub, adding it to the specified model's card.
+
+    Args:
+        repo_id (``str``): Model id from https://hf.co/datasets.
+        task_type (``str``): Task id, refer to
+            https://github.com/huggingface/datasets/blob/master/src/datasets/utils/resources/tasks.json for allowed values.
```

Comment:
This is ok, but just FYI the real source of truth is https://github.com/huggingface/hub-docs/blob/main/js/src/lib/interfaces/Types.ts

Reply:
I might leave it as the JSON file for now because I feel like it's easier to interpret than this one, and (afaik) the JSON's generated automatically from `Types.ts`.

Do you know if we intend on keeping `Types.ts` in the hub-docs repo?